### PR TITLE
CHE-51: Fixing multi-machine workspace creation (e.g. MySql-Java stack)

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Fixes default (MySQL-Java) workspace creation

### What issues does this PR fix or reference?
https://issues.jboss.org/browse/CHE-51
### Previous behavior
Error while creating workspace from multi-machine stack (MySQL - JAVA) 
Openshift service with this name already exist 

### New behavior
WorkspaceID is generated randomly if CHE_WORKSPACE_ID env var does not exist. Running non-dev container as root. For vanilla Che launch (MySQL-JAVA stack) dev-machine UID - 1000, db -0

### CODE REVIEW

**Comment by ibuziuk** [comment]
@l0rd could you please take a look ?

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java [hunk 1]
```diff
@@ -285,7 +292,7 @@ private String createOpenShiftDeploymentConfig(IProject cheProject,
         ((DeploymentConfig) dc).setName(dcName);
         ((DeploymentConfig) dc).setNamespace(cheProject.getName());
 //        ((DeploymentConfig) dc).getNode().get("spec").get("template").get("spec").get("dnsPolicy").set("Default");
-        ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("securityContext").get("runAsUser").set("1000");
+//        ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("securityContext").get("runAsUser").set("1000");
```
Comment (l0rd) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java]: If not needed this line can be removed

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java [hunk 2]
```diff
@@ -311,15 +318,21 @@ private void addContainer(IDeploymentConfig dc, String workspaceID, String image
         dc.getContainer(containerName).setImagePullPolicy(IMAGE_PULL_POLICY_IFNOTPRESENT);
 
         ModelNode dcFirstContainer = ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("containers").get(0);
+        
+        boolean isDevMachine = isDevMachine(exposedPorts);
+        // Run non-dev machine as root user
+        String UID = isDevMachine ? UID_USER : UID_ROOT;
```
Comment (l0rd) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java]: What if it's not a devMachine and the user is not root? I'm thinking about a [multi-machine workspace] where the non-dev machine default user is not root. For these scenarios wouldn't it possible to retrieve the image default user?

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java [hunk 3]
```diff
@@ -311,15 +318,21 @@ private void addContainer(IDeploymentConfig dc, String workspaceID, String image
...
-        dcFirstContainer.get("securityContext").get("runAsUser").set("1000");
+        dcFirstContainer.get("securityContext").get("runAsUser").set(UID);
 
         // LivenessProbe
-        dcFirstContainer.get("livenessProbe").get("tcpSocket").get("port").set(4401);
-        dcFirstContainer.get("livenessProbe").get("initialDelaySeconds").set(120);
-        dcFirstContainer.get("livenessProbe").get("timeoutSeconds").set(1);
+        // TODO: CHE-53 Improve workspaces liveness probe for non-dev machines
+        if (isDevMachine) {
+            dcFirstContainer.get("livenessProbe").get("tcpSocket").get("port").set(4401);
```
Comment (l0rd) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java]: Now that you have defined constant CHE_WORKSPACE_AGENT_PORT you could use it here too.

**Review by l0rd — COMMENTED** [review]
Nice work @ibuziuk! I've added a few comments

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java (hunk 2 above)
Comment (ibuziuk) [reply plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java]: @l0rd thanks for review. For now the logic is the following - all dev mashines are run as user (UID 1000) and all non-dev machines as root (UID 0). Basically, I just checked what UID is used for MySQL-Java stack in by running Che without OpenShift.  I will try to update PR with the logic of getting user from image config

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java (hunk 3 above)
Comment (ibuziuk) [reply plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java]: ok, I will correct this

**Comment by ibuziuk** [comment]
@l0rd PR has been updated

**Review by l0rd — APPROVED** [review]

**Comment by l0rd** [comment]
Thank you @ibuziuk!

**Pull request merged**